keys/keys.go

```diff
@@ -88,6 +92,24 @@ func ImportPrivateKey(privKeyHex string, curve types.CurveType) (*KeyPair, error
 			PublicKey:  pubKey,
 			PrivateKey: rawPrivKey.Seed(),
 		}
+	case types.Secp256r1:
+		crv := elliptic.P256()
+		x, y := crv.ScalarBaseMult(privKey)
+		rawPubKey := ecdsa.PublicKey{X: x, Y: y, Curve: crv}
```

Comment:
Check the `x`,`y` are on the curve and are not zero.

Reply:
fixed

Reply:
Also check that the points x and y are not zero. If they are, IsOnCurve still returns true but we don't want a public key as the identity point since any signature verification will return true.

Reply:
> // A Curve represents a short-form Weierstrass curve with a=-3.
//
// Note that the point at infinity (0, 0) is not considered on the curve, and
// although it can be returned by Add, Double, ScalarMult, or ScalarBaseMult, it
// can't be marshaled or unmarshaled, and IsOnCurve will return false for it.

This is the comment of crypto/elliptic, IsOnCurve will return false.

Reply:
Please check this: https://github.com/golang/go/blob/3298300ddf45a0792b4d8ea5e05f0fbceec4c9f9/src/crypto/elliptic/elliptic.go#L24 @mikebrowncb

Reply:
👋 

I think you meant to tag mikelodder

Reply:
Oh, I'm sorry

Reply:
Seems valid, and it's tested https://github.com/golang/go/blob/3298300ddf45a0792b4d8ea5e05f0fbceec4c9f9/src/crypto/elliptic/elliptic_test.go#L453

Reply:
@LanfordCai One more nit, let's just add a comment before the call to IsOnCurve to explicitly show that this is being checked since it is pretty important to know. Then I will +1

Reply:
@itstehkman I have added comments before each call to IsOnCurve